packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -66,14 +66,29 @@ const DetailView = (props: DetailViewProps) => {
     values.y = values.position.y;
     delete values.position;
 
-    if (values.key !== (form.getValues() || {}).key) {
-      form.resetFields();
-    }
-
     form.setValues(values);
 
   }, [activeSchema, form]);
 
+  useEffect(() => form.resetFields(), [activeSchema.id])
+
+  useEffect(() => {
+    uniqueSchemaKey.current = (value: string): boolean => {
+      for (const s of Object.values(schemas)) {
+        if (s.key === value && s.id !== activeSchema.id) {
+          return false;
+        }
+      }
+      return true;
+    };
+  }, [schemas, activeSchema]);
+
+  const uniqueSchemaKey = useRef((value: string): boolean => true);
+
+  const validateUniqueSchemaKey = (_: any, value: string): boolean => {
+    return uniqueSchemaKey.current(value);
+  }
```

Comment:
This was the only way I could get the `schemas` variable to update.
Otherwise it used a stale value of `schemas` from first page load. I tried wrapping the function in `useCallback` but this didn't work...